segmentstore/storage/src/test/java/io/pravega/segmentstore/storage/chunklayer/ChunkedSegmentStorageTests.java

```diff
@@ -2561,6 +2561,53 @@ private void testParallelReadRequestsOnSingleSegmentWithReentry(int numberOfRequ
         CompletableFuture.allOf(futures).join();
     }
 
+    /**
+     * Test concurrent writes to storage system segments by simulating concurrent writes.
+     *
+     * @throws Exception Throws exception in case of any error.
+     */
+    @Test
+    public void testSystemSegmentConcurrency() throws Exception {
+
+        SegmentRollingPolicy policy = new SegmentRollingPolicy(2); // Force rollover after every 2 byte.
+        @Cleanup
+        TestContext testContext = getTestContext(ChunkedSegmentStorageConfig.DEFAULT_CONFIG.toBuilder().indexBlockSize(3).build());
+        if (!(testContext.metadataStore instanceof InMemoryMetadataStore)) {
+            return;
+        }
+
+        // Simulate some writes to system segment, this should cause some new chunks being added.
+        val n = 4;
+        val data = new byte[n][100];
+
+        var futures = new ArrayList<CompletableFuture<Void>>();
+        @Cleanup("shutdownNow")
+        ExecutorService executor = Executors.newFixedThreadPool(n);
```

Comment:
Just double-checking: is this executor actually needed because we need to have a thread blocked for each write (for testing purposes)? I'm trying to see if the `executorService()` that's already available in this class would be adequate for this test or not.